csharp/src/Microsoft.ML.OnnxTraining/OrtValueCollection.cs

```diff
@@ -0,0 +1,78 @@
+﻿using Microsoft.ML.OnnxRuntime;
+using System;
+using System.Collections.Generic;
+using System.Runtime.InteropServices;
+using System.Text;
+
+namespace Microsoft.ML.OnnxTraining
+{
+    /// <summary>
+    /// The OrtValueCollection holds ort values returned by the error function, but does not actually own any of them
+    /// and therefore does not release them.
+    /// </summary>
+    internal class OrtValueCollection : SafeHandle
```

Comment:
>OrtValueCollection [](start = 19, length = 18)

Exposing native collections to C# is not a good goal

Reply:
In a sense the OrtValueCollection is not a native collection but instead an object that provides access to the collection via its GetAt and SetAt methods.

See lines 280-300 of
https://github.com/MyCaffe/onnxruntime/blob/WinTrainingCSharp/orttraining/orttraining/include/runner/orttraining_c_api.h